src/lib.rs

```diff
@@ -1406,7 +1411,12 @@ impl Build {
         // Target flags
         match cmd.family {
             ToolFamily::Clang => {
-                cmd.args.push(format!("--target={}", target).into());
+                // clang from Android NDK target names are different from rustc.
+                // For example, armv7a-linux-androideabi16-clang passes
+                // --target=armv7a-linux-androideabi16 to clang.
+                if !target.contains("android") || !new_clang_android_compiler(&cmd.path) {
```

Comment:
Can you expand a bit on this? Why does android choke if this `--target` argument is passed?

Reply:
For example:
```
cat ~/Android/Sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi16-clang
#!/bin/bash
if [ "$1" != "-cc1" ]; then
    `dirname $0`/clang --target=armv7a-linux-androideabi16 "$@"
else
    # Target is already an argument.
    `dirname $0`/clang "$@"
fi
```

so final command line generated by scripts looks like
```
clang --target=armv7a-linux-androideabi16  --target=<What cc-rs takes from cargo>
```

So for example definition of `__ANDROID_API__` disappear,
because of clang uses the last `--target=`, not the first one.
That's why I have to re-define `__ANDROID_API__` in the first version of my patch.

Reply:
I added note about "double" `--target` into code's comment section.

Reply:
Could this comment be updated to indicate that the "new" clang compilers automatically pass `--target` so we don't have to?

Reply:
Done and also I changed function name to make it clear.